### ISSUE TITLE
Preview link broken until post is saved for the first time

### ISSUE THREAD

**Issue by hartsook** [WordPress/gutenberg#2544]
## Issue Overview

Gutenberg 0.9
Preview link at top of page does not work unless I update the page first. I get a new browser tab -
 about:blank.  

## Steps to Reproduce (for bugs)

1. Create Gutenberg post
2. add content
3. publish post
4. edit post in Gutenberg
5. make change in post
6. click Preview link at top of page
7. Chrome opens a new tab with about:blank URL
8. This occurs in both author and admin roles
9. If I Update the post first, then the Preview link works.

## Expected Behavior

If I make changes in a post I should be able to preview the post before Updating, especially if this is a Published post and not a Draft post.

Preview link should work like it does currently.

## Current Behavior

Preview link at top of page does not work unless I update the page first. I get a new browser tab about:blank.  

## Todos
- [ ] Tests
- [ ] Documentation

**Comment by youknowriad** [comment]
Duplicate https://github.com/WordPress/gutenberg/issues/1773#issuecomment-316434091 ?

**Comment by karmatosed** [comment]
Despite #1773 being closed, I can still replicate this myself.

**Comment by ptasker** [comment]
Can also replicate this.

**Comment by hedgefield** [comment]
I can replicate this too by opening the demo post on my own site and hitting Preview right away. But that is likely related to [the core issue I reported earlier](https://github.com/WordPress/gutenberg/issues/1761), as I cannot (auto)save a post, so the REST API task label is 👌 .

**Comment by aduth** [comment]
Current status: This was intentionally not implemented because it is not possible yet to support in the REST API.

Related:

- https://github.com/WordPress/gutenberg/issues/1773#issuecomment-316434091
- https://core.trac.wordpress.org/ticket/43316

**Comment by danielbachhuber** [comment]
Looks like this is still an issue:

![preview](https://user-images.githubusercontent.com/36432/40259930-22d84910-5aad-11e8-8841-c4de12fae915.gif)

Notably, it's a bug because the "Preview" button is broken until the first revision is saved. In my GIF, I write "Foo" in the title and then immediately hit the "Preview" button. The first preview UX is broken. However, by the time I navigate back, the first revision is saved and preview begins working correctly.

**Comment by danielbachhuber** [comment]
Likely blocked by #6257

**Comment by danielbachhuber** [comment]
@aduth Can you summarize the underlying issue here, and what needs to be done to address it?

**Comment by aduth** [comment]
@danielbachhuber Se also #7179.

Basically, to be able to complete the redirect, we need a preview link. This is made available for a full save on the post endpoint's `preview_link` property. We don't have this yet for published posts.

In parallel, the [`PostPreviewButton` component](https://github.com/WordPress/gutenberg/blob/423dbed98af82de110af2992806dffff754c8ecb/editor/components/post-preview-button/index.js#L34) should be refactored to rely less on the save process specifically, and merely the presence of this link. I'm working on this currently. Then, once the preview link is added for autosaves (returned by `getEditedPostPreviewLink`), it should "just work".

**Comment by aduth** [comment]
Closed by #7130.
